### PR TITLE
Restructure Analysis Services dataplane cmdlets to use Az module login.

### PR DESCRIPTION
**Pull request by pychuang**
## Description

- Refactored dataplane cmdlets to inherit from a common Analysis Services cmdlet base.
- Added classes to use authentication credentials from the Az module.
- Made Add-AzureASAccount a wrapper of Connect-AzAccount to avoid a breaking change. Add-AzureASAccount is deprecated.
- Removed the implementation of classes used to save Analysis Services independent credentials. Several classes remain existing to avoid a breaking change.
- Updated Add-AzEnvironment and Set-AzEnvironment to accept AzureAnalysisServicesEndpointResourceId to allow test environment log in.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/AnalysisServices/AnalysisServices.Dataplane/Commands/AddAzureASAccount.cs [hunk 1]
```diff
@@ -33,13 +25,14 @@ namespace Microsoft.Azure.Commands.AnalysisServices.Dataplane
...
-    [OutputType(typeof(AsAzureProfile))]
-    public class AddAzureASAccountCommand : AzurePSCmdlet
+    [OutputType(typeof(PSAzureProfile))]
+    public class AddAzureASAccountCommand : AzureRMCmdlet
     {
         private const string UserParameterSet = "UserParameterSetName";
         private const string ServicePrincipalWithPasswordParameterSet = "ServicePrincipalWithPasswordParameterSetName";
         private const string ServicePrincipalWithCertificateParameterSet = "ServicePrincipalWithCertificateParameterSetName";
 
+        // This is ignored sicne we only supported public cloud
```
Comment (gnair-ms) [inline src/AnalysisServices/AnalysisServices.Dataplane/Commands/AddAzureASAccount.cs]: typo: since..

**Comment by pychuang** [comment]
Need to test the cmdlets with ":rw" in instance name

**Comment by pychuang** [comment]
> Need to test the cmdlets with ":rw" in instance name

Tested. Sync works.

File: src/Accounts/Accounts/Environment/AddAzureRMEnvironment.cs [hunk 2]
```diff
@@ -163,6 +163,10 @@ public string DataLakeAudience
            HelpMessage = "The endpoint to use when communicating with the Azure Log Analytics API.")]
```
Comment (markcowl) [inline src/Accounts/Accounts/Environment/AddAzureRMEnvironment.cs]: Need to add a line in the changelog.md for Az.Accounts to describe this change - thsi is the trigger for changing the version and releasing a new version fo the module

Comment (pychuang) [reply src/Accounts/Accounts/Environment/AddAzureRMEnvironment.cs]: Updated ChangeLogs

**Pull request merged**